### PR TITLE
7. POST /api/reviews/:review_id/comments

### PR DESCRIPTION
**Pull request by t-cbailey**
7. POST /api/reviews/:review_id/comments

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -84,4 +84,65 @@ describe("App", () => {
...
+      morenonsense: 1,
+    };
+    return request(app)
+      .post("/api/reviews/1/comments")
+      .send(commentData)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Unsupported body format");
+      });
+  });
```
Comment (cjplatten) [inline __tests__/app.test.js:188]: Great start! I'd like you to think of other tests - what if the submitted object has unnecessary proerties? what if the review id is invalid?

File: playground.sql [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
-WHERE review_id = 1;
\ No newline at end of file
+
+INSERT INTO comments (author, body) VALUES ($1, $2) RETURNING *;
+
+
+-- SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.review_id FROM reviews 
+--   JOIN comments ON comments.review_id = reviews.review_id
+--   WHERE reviews.review_id = 2
+--   ORDER BY comments.created_at;
```
Comment (cjplatten) [inline playground.sql:9]: git ignore your playground files as they are not part of the working code.

**Review by cjplatten — COMMENTED** [review]
Looking good so far, please see comments

File: __tests__/app.test.js [hunk 3]
```diff
@@ -103,46 +103,87 @@ describe("App", () => {
...
+
+    test("POST 404- review_id invalid", () => {
+      const commentData = {
+        username: "mallionaire",
+        body: "my review text",
+      };
+      return request(app)
+        .post("/api/reviews/nonsense/comments")
+        .send(commentData)
+        .expect(404)
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:142]: Invalid ids should be 400 not 404.

File: models.js [hunk 4]
```diff
@@ -21,13 +21,21 @@ exports.findReviewByID = (id) => {
...
-      `INSERT INTO comments (author, body, review_id) VALUES ($1, $2, $3) RETURNING *`,
-      [username, body, review_id]
-    )
-    .then((res) => {
-      return res.rows;
-    });
+  if (
+    !data.hasOwnProperty("username") ||
+    !data.hasOwnProperty("body") ||
+    typeof data.body != "string" ||
```
Comment (dappernerddesigns) [inline models.js:27]: Everything coming back from psql will be a string, even if you send a json body of {"body": 2}  I get that you're checking for data types, but the table constraints in the seed file will do this for you. If the body column is a text or varchar it is going to kick off at an integer.

**Review by dappernerddesigns — APPROVED** [review]
A few little bits I'd like you to tighten up before you move on. An invalid id will always be an error code of 400 as we're expecting a number and got text (in your test). Double check your white listing for checking data types. You should already have a psql error handler that will take care of it.

**Comment by t-cbailey** [comment]
Thanks for the feedback Verity, but I’m confused by this- “Double check your white listing for checking data types. You should already have a psql error handler that will take care of it.” Do you mean I already do have one, or I should have already written one?

**Pull request closed without merging**